### DISCUSSION TITLE
Wavy laser engraved lines

### DISCUSSION THREAD

**Discussion by teamtexas55** [arkypita/LaserGRBL#954] [Q&A]
I'm just purchased an  Ortur 15 watt laser and have downloaded GBRL and updated the firm ware to 1.34 version.  When I run an engraving file that has curved lines I'm getting a wavy line instead of a smooth curve.  It doesn't happen everywhere though  Dan
![Wavy cut](https://user-images.githubusercontent.com/63623928/79177383-54ed7200-7dc8-11ea-98b6-64e661ee3881.JPG)

**Comment by gmmanonymus111** [comment]
My suggestion in the first place is to check if every screw is tight enough. Loose screws might cause issues like this.

**Comment by teamtexas55** [comment]
Thanks Alex.  I did check both drive bands and adjusted them. I will double check for loose screws.  I tested a few more times and have found the wavy lines always come back in the exact same spots. Makes me think it’s a program glitch.

Sent from my iPhone

On Apr 15, 2020, at 6:12 AM, Alex <notifications@github.com<mailto:notifications@github.com>> wrote:


My suggestion in the first place is to check if every screw is tight enough. Loose screws might cause issues like this.

**Comment by gmmanonymus111** [comment]
Which mode do you use to create the engrave-image in LaserGRBL? Also, could you upload the original image?

**Comment by teamtexas55** [comment]
Here are some screen shots of the settings I used and the original image in jpeg.

![Old Ash wine uke logo black, white and grey](https://user-images.githubusercontent.com/63623928/79342970-7139fd80-7ef3-11ea-88e5-df3b9843c237.jpg)
![IMG_4006](https://user-images.githubusercontent.com/63623928/79343088-9595da00-7ef3-11ea-9268-f96ee4c24b07.JPG)
![IMG_4007](https://user-images.githubusercontent.com/63623928/79343091-962e7080-7ef3-11ea-88c6-beb963824202.JPG)
![IMG_4005](https://user-images.githubusercontent.com/63623928/79343095-975f9d80-7ef3-11ea-8d58-fd95bcbeff19.JPG)

**Comment by gmmanonymus111** [comment]
For me, it produces good results, not as wavy as yours.
![wine](https://user-images.githubusercontent.com/9658243/79361715-f9e18980-7f45-11ea-8736-dd2c836fa8d9.png)
Excuse the poor wood quality

- The first one is made with `Smooth (HQ Bicubic)`, this looks way better.
- The second one is made with `Sharp (NearestNeighbor)` like yours, if you look closely, there is a small wavyness(?) in it, but not that noticable compared to yours.

You should give it a try with `Smooth (HQ Bicubic)` to see if it gets better.
What version of LaserGRBL and GRBL are you using? I can't quite read it, because your images are too small, and I don't recall GRBL has 1.34 version (only 0.9 or 1.1)

EDIT: 1.34 is Ortur's firmware version, which might work differently than standard GRBL on Arduino (or any Atmega chip)

**Comment by teamtexas55** [comment]
It does look better on the 1st one.  The version is GBRL 3.1.2.  I will give it a try on Smooth and see what it does.
________________________________

**Comment by teamtexas55** [comment]
I ran the file again after I updated the firm ware on the laser to 1.34 version.  In smooth this is my result. 
![IMG_4009](https://user-images.githubusercontent.com/63623928/79398554-94da6380-7f46-11ea-8293-0a273511970a.JPG)

This is reduced in size a bit, but seems much smoother than it did before I updated the firmware.  This was run at 3000 mm/sec speed and 750 Power.   I left the brightness, contrast slides at the default.

**Comment by gmmanonymus111** [comment]
Looks better. Could you post your GRBL settings (by sending `$$` command to GRBL)?
You could also try with lower speed, as belts tend to stretch on high load.

**Comment by arkypita** [comment]
I think it has too high acceleration and maximum speed values, so when the head moves quickly for jumping empty spaces the machine and the motors cannot do it with sufficient precision.

I am talking about this parameters

![image](https://user-images.githubusercontent.com/8782035/79429159-74062280-7fc7-11ea-802c-137158a403ab.png)

Check yours, write them down somewere (or export to file) and try set lower values.

**Comment by teamtexas55** [comment]
Here are the settings. 
![IMG_4024](https://user-images.githubusercontent.com/63623928/79677912-7a5cf000-81bb-11ea-958d-dca2514908f0.JPG)
![IMG_4025](https://user-images.githubusercontent.com/63623928/79677913-7b8e1d00-81bb-11ea-89d5-e34b561d7c49.JPG)
![IMG_4026](https://user-images.githubusercontent.com/63623928/79677914-7cbf4a00-81bb-11ea-8fab-406c72e1123c.JPG)

What do you think?

**Comment by gmmanonymus111** [comment]
Sorry @teamtexas55, but your images are too small to read.

**Comment by teamtexas55** [comment]
Lets try that again.
![IMG_4024 (1)](https://user-images.githubusercontent.com/63623928/79694431-d1071000-8235-11ea-9e47-45dbd6e73335.JPG)
![IMG_4025 (1)](https://user-images.githubusercontent.com/63623928/79694454-eaa85780-8235-11ea-9353-650124132720.JPG)
![IMG_4026 (1)](https://user-images.githubusercontent.com/63623928/79694472-027fdb80-8236-11ea-9ca2-d1b3079c58e7.JPG)

**Comment by gmmanonymus111** [comment]
Looks much better now!
Everything seems fine, except X & Y acceleration (`$120`, `$121`), which seems too high for me.
My machine has 25 mm/sec^2, but it is a leadscrew-driven one, so your belt driven one can go faster obviosuly, although `2200` and `1800 `is a bit much. Try to lower it to `1320 `& `1080 `(60%) and check if it gets any better. Lower again, if needed.
Maybe someone with more experience with belt-driven engravers could suggest better values.

Junction deviation (`$11`) is set to `0.010` for me. This may also affect curves.

My step pulse time (`$0`) and step idle delay (`$1`) are half of yours, but it should not cause any problems the way it engraves.
